File: C++/507. Perfect Number.cpp
class Solution {
public:
    bool checkPerfectNumber(int num) {
        if(num <= 1) return false;
        int ans = 1;
        for(int i=2;i*i<=num;i++)
        {
            if(num%i==0)
                ans += i + num/i;
        }
        return num==ans;
    }
};

File: Python/507. Perfect Number.py
class Solution:
    def checkPerfectNumber(self, num):
        """
        :type num: int
        :rtype: bool
        """
        if num <= 1:
            return False
        i, end = 2, int(math.sqrt(num))
        ans = 1
        while i <= end:
            if num % i == 0:
                ans += i + num / i
            i += 1
        return ans == num